experimental/bridge/opentracing/lib.go

```diff
@@ -0,0 +1,399 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opentracing
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"google.golang.org/grpc/codes"
+
+	ot "github.com/opentracing/opentracing-go"
+	otext "github.com/opentracing/opentracing-go/ext"
+	otlog "github.com/opentracing/opentracing-go/log"
+
+	otelcore "go.opentelemetry.io/api/core"
+	otelevent "go.opentelemetry.io/api/event"
+	otelregistry "go.opentelemetry.io/api/registry"
+	oteltrace "go.opentelemetry.io/api/trace"
+)
+
+type bridgeSpanContext struct {
+	// TODO: have a look at the java implementation of the shim to
+	// see what do they do with the baggage items
+	baggageItems    map[string]string
+	otelSpanContext otelcore.SpanContext
+}
+
+var _ ot.SpanContext = &bridgeSpanContext{}
+
+func (c *bridgeSpanContext) ForeachBaggageItem(handler func(k, v string) bool) {
+	for k, v := range c.baggageItems {
+		if !handler(k, v) {
+			break
+		}
+	}
+}
+
+func (c *bridgeSpanContext) setBaggageItem(restrictedKey, value string) {
+	if c.baggageItems == nil {
+		c.baggageItems = make(map[string]string)
+	}
+	c.baggageItems[restrictedKey] = value
+}
+
+func (c *bridgeSpanContext) baggageItem(restrictedKey string) string {
+	return c.baggageItems[restrictedKey]
+}
+
+type bridgeSpan struct {
+	otelSpan oteltrace.Span
+	ctx      *bridgeSpanContext
+	tracer   *bridgeTracer
+}
+
+var _ ot.Span = &bridgeSpan{}
+
+func (s *bridgeSpan) Finish() {
+	s.otelSpan.Finish()
+}
+
+func (s *bridgeSpan) FinishWithOptions(opts ot.FinishOptions) {
+	// FIXME: I can't see how we can support the FinishTime option.
+	//
+	// TODO: Do we handle LogRecords by calling Event for all the
+	// items in the array? Same with BulkLogData?
+	s.Finish()
```

Comment:
The FIXME: We should add a `FinishOptions` option, or similar. See the discussion around https://github.com/open-telemetry/rfcs/blob/master/text/0002-remove-spandata.md, which requires a way to set the finish timestamp similar to OpenTracing.

The TODO: I think so. There are some inconsistencies in the current Event API that should be cleaned up. TODO(jmacd): I'll update a github issue to explain this comment.